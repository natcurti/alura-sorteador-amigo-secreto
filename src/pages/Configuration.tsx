import Card from "../components/Card";
import Footer from "../components/Footer";
import Form from "../components/Form";
import ParticipantsList from "../components/ParticipantsList";

const Configuration = () => {
  return (
    <Card>
      <section>
        <h2>Vamos come√ßar!</h2>
        <Form />
        <ParticipantsList />
        <Footer />
      </section>
    </Card>
  );
};

export default Configuration;
