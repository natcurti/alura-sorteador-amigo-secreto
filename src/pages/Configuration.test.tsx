import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Configuration from "./Configuration";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigate,
  };
});
describe("A página de configuração", () => {
  test("deve ser renderizada corretamente", () => {
    const { container } = render(
      <RecoilRoot>
        <Configuration />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
