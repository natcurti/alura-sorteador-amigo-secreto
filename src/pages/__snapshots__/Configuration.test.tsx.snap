// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Vamos começar!
      </h2>
      <form>
        <div
          class="group-input-btn"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            value=""
          />
          <button
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="footer-config"
      >
        <button
          class="button"
          disabled=""
        >
          Iniciar brincadeira
        </button>
      </footer>
    </section>
  </div>
</div>
`;
